clarity_ext/repository/step_repository.py

```diff
@@ -169,3 +173,168 @@ def get_process_type(self):
     def get_process(self):
         """Returns the currently running process (step)"""
         return self.session.current_step
+
+
+class WrappablePairs(object):
+    """
+    Contains all artifact pairs within a step, containing genologics Artifact instances
+    These are in turn ready to be wrapped into clarity-ext domain instances
+    Usage:
+    wrappable_pairs = WrappablePairs(session, input_output_maps, mode=<stateful | stateless>)
+    wrappable_pairs.validate()  # compares xml for stateful and stateless
+    for input, output, output_generation_type in wrappable_pairs:
+        input  # genologics Artifact, either stateful or stateless
+        output  # genologics Artifact
+
+    The validation compares the stateless and statefull xml representation,
+    and discards any discrepancies in the qc-flag
+    """
+
+    MODE_STATELESS = "stateless"
+    MODE_STATEFUL = "stateful"
+
+    def __init__(self, session, input_output_maps, state_mode):
+        # Note, the force flag should be set to False as soon as the test period is over!
```

Comment:
Cool :) I think we should even go further and just remove all the state-related code when it's over.